src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/ComparisonExpression.java

```diff
@@ -19,9 +17,29 @@
  * {"username" : {"$eq" : "aaron"}} In here we expand the shortcut into a canonical long form, so it
  * is all the same.
  */
-public record ComparisonExpression(
-    @NotBlank(message = "json node path can not be null in filter") String path,
-    @Valid @NotEmpty List<FilterOperation<?>> filterOperations) {
+public class ComparisonExpression {
+
+  @NotBlank(message = "json node path can not be null in filter")
```

Comment:
If changing record into class, you probably need to move validation annotations to constructor parameters for them to work, due to differences in how records work compared to regular POJOs.
Not sure if we have tests verifying that validation occurs or not, that should tell whether using annotations on fields work.